### ISSUE TITLE
As an app developer I want to show the display value as a variable in text

### ISSUE THREAD

**Issue by Febakke** [Altinn/app-frontend-react#1105]
### Description

When I want to present data that has been filled out earlier in the app to the user again, the variable that i get from the datamodel does not make sense when it is from an option or any other value that has a different display value then the one in the datamodel. 

**Why?**
There are many cases where you want to use data that has been prevously filled inn by the user to remind them or give them some context to what their task is on the page. It is recomended to show read-only data as plain text. This will also give the app-developer the possibility to style this element as they want. 

**Other problems**
This issue came up because our dropdown do not have a read-only state. This should be [fixed in the design system.](https://github.com/digdir/designsystem/issues/307)
But this means that there are no way to show this kind of data in an accesible way in our solution today. 

### Additional Information

This issue is related to #207 and is also a required to solve if we want to be able to offer this functionality to our users

**Comment by bjosttveit** [comment]
What if you could use expressions to build texts instead of having variables in the text resource itself? I just looked at the code that deals with variables in text and it is a mess of spaghetti code. If we simply add a new function to expressions that will look up a text resource and give you the value, as well as a function for getting a components display value, something like this should be possible:

```json
{
  "textResourceBindings": {
    "title": ["concat", ["text", "dateSent"], ["displayValue", "date-sent-picker"]]
  }
}
```

This should also be relatively straight forward to implement as we already have displayValue for all component classes readily available in the node hierarchy.

That would greatly simplify the logic in app-frontend as the expression engine would be doing the heavy lifting, and altinn studio would only need to maintain one GUI for expressions instead of an additional one for variables in text.

**Comment by bjosttveit** [comment]
> What if you could use expressions to build texts instead of having variables in the text resource itself? I just looked at the code that deals with variables in text and it is a mess of spaghetti code. If we simply add a new function to expressions that will look up a text resource and give you the value, as well as a function for getting a components display value, something like this should be possible:
> 
> ```json
> {
>   "textResourceBindings": {
>     "title": ["concat", ["text", "dateSent"], ["displayValue", "date-picker-component"]]
>   }
> }
> ```
> 
> This should also be relatively straight forward to implement as we already have displayValue for all component classes readily available in the node hierarchy.
> 
> That would greatly simplify the logic in app-frontend as the expression engine would be doing the heavy lifting, and altinn studio would only need to maintain one GUI for expressions instead of an additional one for variables in text.

This is now possible since #1341
